state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -1006,7 +936,239 @@ fn witness_broadcast(tx_out_id: [u8; 4]) {
+
+			broadcast_id
+		})
+		.then_execute_at_block(expiry, |broadcast_id| broadcast_id)
+		.then_execute_with(|broadcast_id| {
+			// Broadcast should be aborted
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastAborted { broadcast_id },
+			));
+			assert!(AbortedBroadcasts::<Test, Instance1>::get().contains(&broadcast_id));
+			assert_eq!(FailedBroadcasters::<Test, Instance1>::decode_len(broadcast_id), None);
+		});
+}
+
+#[test]
+fn aborted_broadcasts_can_still_succeed() {
+	let mut expiry = 0;
+	new_test_ext()
+		.execute_with(|| {
+			let (broadcast_id, transaction_out_id) = start_mock_broadcast();
+			expiry = System::block_number()
+				.saturating_add(<Test as crate::Config<Instance1>>::BroadcastTimeout::get());
+			ready_to_abort_broadcast(broadcast_id);
+
+			(broadcast_id, transaction_out_id)
+		})
+		.then_execute_at_block(expiry, |(broadcast_id, transaction_out_id)| {
+			(broadcast_id, transaction_out_id)
+		})
+		.then_execute_with(|(broadcast_id, transaction_out_id)| {
+			// Broadcast should be aborted
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastAborted { broadcast_id },
+			));
+			assert_eq!(FailedBroadcasters::<Test, Instance1>::decode_len(broadcast_id), None);
+			assert!(AbortedBroadcasts::<Test, Instance1>::get().contains(&broadcast_id));
+
+			// Broadcast can still be reported as successful
+			assert_ok!(Broadcaster::transaction_succeeded(
+				RuntimeOrigin::root(),
+				transaction_out_id,
+				Default::default(),
+				ETH_TX_FEE,
+				MOCK_TX_METADATA,
+			));
+
+			// Storage should be cleaned, event emitted.
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastSuccess {
+					broadcast_id,
+					transaction_out_id,
+				},
+			));
+			assert_broadcast_storage_cleaned_up(broadcast_id);
+		});
+}
+
+#[test]
+fn broadcast_retry_delay_works() {
+	let mut target = 0;
+	let mut broadcast_id = 0;
+	let mut tx_out_id = Default::default();
+	new_test_ext()
+		.execute_with(|| {
+			(broadcast_id, tx_out_id) = start_mock_broadcast();
+
+			BroadcastDelay::set(None);
+			// With no delay, retries are added to the normal queue, and is retried in the next
+			// block.
+			assert_ok!(Broadcaster::transaction_failed(RuntimeOrigin::signed(0u64), broadcast_id));
+			assert_eq!(BroadcastRetryQueue::<Test, Instance1>::get()[0].broadcast_id, broadcast_id);
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastRetryScheduled {
+					broadcast_id,
+					retry_block: System::block_number() + 1,
+				},
+			));
+		})
+		.then_execute_at_next_block(|_| {})
+		.then_execute_with(|_| {
+			BroadcastDelay::set(Some(10));
+			// Set delay - retries will be added to the Delayed queue.
+			assert_ok!(Broadcaster::transaction_failed(RuntimeOrigin::signed(1u64), broadcast_id));
+			target = System::block_number() + 10;
```

Comment:
same here, let's use a const instead of passing 10 into both places, otherwise it's not super clear they're related